### ISSUE TITLE
Dynamic parameters for the sensor class

### ISSUE THREAD

**Issue by abassign** [JSBSim-Team/jsbsim#133]
Often the sensors have problems, for example the lack of power supply, at this point it would be useful to be able to insert directly into the sensor its operating status (for example in the form of a gain parameter that is not constant, but associated with a parameter), it would be useful also to be able to have a sensor value when it is not working, also in the form of a possibly recursive parameter (for example the value of the sensor itself when it is placed in an inoperative state). Also the error parameter should be a parameter to be able to simulate the degradation of the sensor for various causes, for example an excessive acceleration etc ...

**Comment by bcoconni** [comment]
Yes, this will be part of some refactoring of the gain/bias/delay feature. At the moment, there is a lot of code duplication to handle these features in the sensors, I will take this opportunity to allow "dynamic" parameters for them.

**Comment by abassign** [comment]
@bcoconni Thanks for your interest, I believe that the possibility of extending variable parameters over time is very useful to simplify the code and make simulation more realistic.

**Comment by bcoconni** [comment]
> Thanks for your interest, I believe that the possibility of extending variable parameters over time is very useful to simplify the code and make simulation more realistic.

No problem but don't expect it to happen any time soon as I have already plenty on my plate.

**Comment by legoboyvdlp** [comment]
Just on the topic of sensors, I wonder if someone could explain what the value in `<bits> number </bits>` means; what does this number control?

**Comment by seanmcleod** [comment]
@legoboyvdlp the `bits` field if specified is used to quantize the output, so for example if you specify `10 bits` then you will only see 2^10 = 1024 distinct values output, so in effect limiting the resolution you will see from the sensor.

**Comment by legoboyvdlp** [comment]
Thank you!
